Fix Issue #56: Updates to support Ubuntu-22 and GCC11

manthonyaiello opened the pull request:
Fixes Issue #56 

GCC 11 changes the header file for `std::numerical_limits`; this fix
addresses that change.

gcov with GCC 11 drops `__gcov_flush` and recommends the use of the
`__gcov_dump` instead.

The `bootstrap.patch` is a long-unused file.

The GitHub workflows are all updated to include Ubuntu-22 in the matrix
builds. We should remove Ubuntu-18 in a future issue.

acfoltzer commented:
This now builds three ways: Linux native, Max native, and Linux + Docker.

And good news, the AutomationRequestTest failure seems to just be happening because its runtime is usually around 25-30 seconds, but Meson's default test timeout is 30 seconds. By changing the way the scripts invoke the Meson tester, the timeout is now 60 seconds.

acfoltzer commented:
@artimid, did you mean to close this without merging?

manthonyaiello committed:
Updates to support Ubuntu-22 and GCC11

GCC 11 changes the header file for `std::numerical_limits`; this fix
addresses that change.

gcov with GCC 11 drops `__gcov_flush` and recommends the use of the
`__gcov_dump` instead.

The `bootstrap.patch` is a long-unused file.

The GitHub workflows are all updated to include Ubuntu-22 in the matrix
builds. We should remove Ubuntu-18 in a future issue.
